fw/rbcx-coprocessor/include/Bsp.hpp

```diff
@@ -122,29 +128,45 @@ inline bool pinRead(PinDef pin) {
 }
 
 inline void pinWrite(PinDef pin, bool value) {
-    HAL_GPIO_WritePin(pin.first, pin.second, value ? GPIO_PIN_SET : GPIO_PIN_RESET);
+    HAL_GPIO_WritePin(
+        pin.first, pin.second, value ? GPIO_PIN_SET : GPIO_PIN_RESET);
 }
 
-inline void pinToggle(PinDef pin) {
-    HAL_GPIO_TogglePin(pin.first, pin.second);
-}
+inline void pinToggle(PinDef pin) { HAL_GPIO_TogglePin(pin.first, pin.second); }
 
 inline void pinsInit() {
     pinInit(ledPins, GPIO_MODE_OUTPUT_PP, GPIO_NOPULL, GPIO_SPEED_FREQ_LOW);
 
     pinWrite(powerPin, 1);
     pinInit(powerPin, GPIO_MODE_OUTPUT_OD, GPIO_NOPULL, GPIO_SPEED_FREQ_LOW);
 
-    for (auto button: buttonPin)
-       pinInit(button, GPIO_MODE_INPUT, GPIO_PULLUP, GPIO_SPEED_FREQ_LOW);
+    for (auto button : buttonPin)
+        pinInit(button, GPIO_MODE_INPUT, GPIO_PULLUP, GPIO_SPEED_FREQ_LOW);
 
     // USB
     pinInit(usbDnPin, GPIO_MODE_AF_PP, GPIO_NOPULL, GPIO_SPEED_FREQ_HIGH);
     pinInit(usbDpPin, GPIO_MODE_AF_PP, GPIO_NOPULL, GPIO_SPEED_FREQ_HIGH);
-    pinInit(usbDpPullUpPin, GPIO_MODE_OUTPUT_PP, GPIO_NOPULL, GPIO_SPEED_FREQ_LOW);
-    pinInit(usbBusDetectionPin, GPIO_MODE_INPUT, GPIO_NOPULL, GPIO_SPEED_FREQ_LOW);
+    pinInit(
+        usbDpPullUpPin, GPIO_MODE_OUTPUT_PP, GPIO_NOPULL, GPIO_SPEED_FREQ_LOW);
+    pinInit(
+        usbBusDetectionPin, GPIO_MODE_INPUT, GPIO_NOPULL, GPIO_SPEED_FREQ_LOW);
 }
 
 inline bool isPressed(PinDef button) {
     return button == buttonOnPin ? pinRead(button) : !pinRead(button);
-}
\ No newline at end of file
+}
+
+inline uint32_t getButtons() {
+    uint32_t buttons = 0;
+    for (size_t i = 0; i < buttonPin.size(); ++i) {
```

Comment:
Dotaz:
Neměli bychom tady to pole `buttonPin` procházet od konce?

Protože jestli to dobře chápu, tak pin s indexem 0 bude nejvíc v levo (2^5) a pin s indexem 5 bude nejvíc v pravo (2^0).

Připadá mi, že u `setLeds()` je to obráceně a za mě správně. Nebo se pletu?

Reply:
dík najs opraveno